### PR TITLE
Clean up fail fast on not found logic

### PR DESCRIPTION
**Pull request by medb**
This change also removes extra `getItemInfo` call in gRPC read channel

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java [hunk 1]
```diff
@@ -752,25 +745,16 @@ private SeekableByteChannel open(
...
-              storageStubProvider,
-              itemInfo,
-              watchdog,
-              metricsRecorder,
-              readOptions,
-              BackOffFactory.DEFAULT,
-              this.storageOptions);
+      return new GoogleCloudStorageGrpcReadChannel(
+          storageStubProvider,
+          itemInfo == null ? getItemInfo(resourceId) : itemInfo,
```
Comment (abmodi) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:752]: This PR is just cleaning up the metadata call but not adding support for failFast logic for gRPC. Is my understanding correct?

Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:752]: Yes, I think that gRPC channel as it written right now always requires metadata call/info - this can be improved as follow up, if possible.

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java [hunk 2]
```diff
@@ -177,8 +172,8 @@ public GoogleCloudStorageReadChannel(
 
     // Initialize metadata if available.
     GoogleCloudStorageItemInfo info = getInitialMetadata();
-    if (info != null) {
-      initMetadata(info);
+    if (info != null || readOptions.isFastFailOnNotFoundEnabled()) {
```
Comment (singhravidutt) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java:175]: Why is this required? IIUC, `info` can only be `null` iff fastFail is `disabled`. If `fastFail` is `enabled` then either client will get a `FileNotFoundException` or channel with NOT NULL `info`.

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java [hunk 3]
```diff
@@ -225,7 +212,7 @@ ExponentialBackOff createBackOff() {
    */
   @Nullable
   protected GoogleCloudStorageItemInfo getInitialMetadata() throws IOException {
```
Comment (singhravidutt) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java:214]: Is there any implication of removing this all together?

Comment (singhravidutt) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java:214]: nvm, this function is getting override too in `GoogleCloudStorageImpl.java`

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java [hunk 4]
```diff
@@ -716,14 +716,7 @@ public SeekableByteChannel open(
     checkArgument(
         resourceId.isStorageObject(), "Expected full StorageObject id, got %s", resourceId);
 
-    // The underlying channel doesn't initially read data, which means that we won't see a
```
Comment (singhravidutt) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:719]: Thank you for cleaning this up.

**Review by singhravidutt — APPROVED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java (hunk 4 above)
Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java:719]: np
